### PR TITLE
Pass `ModelArray` to `Xios.write`, rather than array of `double`s

### LINKED ISSUE #544: Pass `ModelArray` to `Xios.write` rather than array of `double`s
The newly introduced `Xios` class has a `write` method that accepts a C-style array of `double`s, along with its dimensions in each direction. It would be more in line with Nextsim's approach to instead pass a `ModelArray`, which already knows about both the data and dimensions.

### PR DESCRIPTION
**Pull request by jwallwork23**
# Pass `ModelArray` to `Xios.write`, rather than array of `double`s
### Task List
- [x] Defined the tests that specify a complete and functioning change (*It may help to create a [design specification & test specification](../../../wiki/Specification-Template)*)
- [x] Implemented the source code change that satisfies the tests
- [x] Documented the feature by providing worked example
- [x] Updated the README or other documentation
- [x] Completed the pre-Request checklist below

---
# Change Description

In the existing implementation of the `Xios` class, its `write` method accepts a C-style array of `double`s, along with its dimensions in each direction. The approach adopted here - better in line with neXtSIM-DG's approach - is to instead pass a `ModelArray`, which knows about both the data and dimensions.

NOTE: #559 will need to be merged first.

---
# Test Description

Updated the tests accordingly.

- [x] Implement changes in XIOS integration test.
- [x] Fix linking/build for XIOS integration test.

---
### Pre-Request Checklist

- [x] The requirements of this pull request are fully captured in an issue or design specification and are linked and summarised in the description of this PR
- [x] No new warnings are generated
- [x] The documentation has been updated (or an issue has been created to track the corresponding change)
- [x] Methods and Tests are commented such that they can be understood without having to obtain additional context
- [x] This PR/Issue is labelled as a bug/feature/enhancement/breaking change
- [x] File dates have been updated to reflect modification date
- [x] This change conforms to the conventions described in the README

### CODE REVIEW

**Comment by jwallwork23** [comment]
This is now ready for review. The issue was a missing `Nextsim::` when defining the `ParametricMesh`.

File: core/test/XiosWrite_test.cpp [hunk 1]
```diff
@@ -102,37 +105,47 @@ MPI_TEST_CASE("TestXiosWrite", 2)
...
+    HField field_3D(ModelArray::Type::Z);
+    field_3D.resize();
+    for (size_t k = 0; k < n3; ++k) {
+        for (size_t j = 0; j < n2; ++j) {
+            for (size_t i = 0; i < n1; ++i) {
+                field_3D(i, j, k) = 1.0 * (i + n1 * (j + n2 * k));
+            }
+        }
     }
+    // TODO: field_4D?
```
Comment (jwallwork23) [inline core/test/XiosWrite_test.cpp:130]: I'd taken a note that we should support I/O 2D, 3D, and 4D fields. However, looking at https://github.com/nextsimhub/nextsimdg/blob/main/core/src/discontinuousgalerkin/ModelArrayDetails.cpp I only see `ModelArray`s with 2 or 3 components. @einola @timspainNERSC please could you remind me what the use case is for 4D?

**Review by timspainNERSC — APPROVED** [review]
Either add the four dimensional write test now, otherwise I will add it when I get to adding DG components to the T_ice field. Other than that, this all looks good.

**Comment by jwallwork23** [comment]
> Either add the four dimensional write test now, otherwise I will add it when I get to adding DG components to the T_ice field. Other than that, this all looks good.

If it's okay, I'll leave it to you to as I'm unsure how all the model array details would look. Thanks!

**Comment by timspainNERSC** [comment]
> > Either add the four dimensional write test now, otherwise I will add it when I get to adding DG components to the T_ice field. Other than that, this all looks good.
> 
> If it's okay, I'll leave it to you to as I'm unsure how all the model array details would look. Thanks!

Me neither! (at the moment)

**Pull request merged**